src/core/mesh/qgsmeshlayer.h

```diff
@@ -523,6 +523,22 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer, public QgsAbstractProfileSo
       */
     QgsMeshDatasetValue datasetValue( const QgsMeshDatasetIndex &index, const QgsPointXY &point, double searchRadius = 0 ) const;
 
+    /**
+      * Interpolates the value on the given point from given dataset, even if the layer has not been rendered previously.
+      *
+      * \note This will create the mesh if necessary before calling datasetValue().
+      * \see datasetValue
+      *
+      * \param renderContext the QgsRenderContext used by QgsRenderer
+      * \param index dataset index specifying group and dataset to extract value from
+      * \param point point to query in map coordinates
+      * \param searchRadius the radius of the search area in map unit
+      * \returns interpolated value at the point returned by datasetValue()
+      *
+      * \since QGIS 3.42
+      */
+    QgsMeshDatasetValue datasetValueUncached( const QgsRenderContext &renderContext, const QgsMeshDatasetIndex &index, const QgsPointXY &point, double searchRadius = 0 );
```

Comment:
It's not clear to me why we need the new method and when this should be used.
Could we not conditionally call `updateTriangularMesh( renderContext.coordinateTransform() );` before calling `featureInfoFromMeshLayer()` in wmsrenderer instead?